hashintel/hash
# Entity editor links table design & functionality

## 🌟 What is the purpose of this PR?

This PR introduces updates the design of the links table, and adds the following functionality to links table
- remove link
- add link
- change link

Note that links with `maxItems > 1` is not supported yet. So only single links are supported, not link arrays. There will be follow-up PRs. 

## 🔗 Related links

- [Design](https://www.figma.com/file/gydVGka9FjNEg9E2STwhi2/HASH-App?node-id=6468%3A253538&t=5aqL5g0lLyxTzeyF-4) _(internal)_


## 🔍 What does this change?

- 

## ⚠️ Known issues

- [periodic ‘must have right endpoint’ crash when repeatedly archiving and creating links to replace a link](https://hashintel.slack.com/archives/C022217GAHF/p1669893629364659?thread_ts=1669891299.452099&cid=C022217GAHF) _(internal)_

## 🐾 Next steps

Implement list of links when `maxItems > 1` on a link


## ❓ How to test this?



1.  Checkout the branch / view the deployment
1.  Create an entity with links (you can use `create demo entity type` button here https://github.com/hashintel/hash/commit/bfa2f02480b29584944c48ced897a9c3ecd92458)
1.  Confirm that entity editor links table has the functionality shown in the video below at demo section

## 📹 Demo


https://user-images.githubusercontent.com/39553853/205078310-8e332afa-a886-495a-b888-31b815b5d1c4.mov
